src/lib/renderElement.js

```diff
@@ -3,8 +3,20 @@ import { createElement } from "./createElement";
 import { normalizeVNode } from "./normalizeVNode";
 import { updateElement } from "./updateElement";
 
+const nodeToContainerMap = new WeakMap();
+
 export function renderElement(vNode, container) {
+  vNode = normalizeVNode(vNode);
+
   // 최초 렌더링시에는 createElement로 DOM을 생성하고
-  // 이후에는 updateElement로 기존 DOM을 업데이트한다.
+  if (!nodeToContainerMap.has(container)) {
+    container.append(createElement(vNode));
+  } else {
+    // 이후에는 updateElement로 기존 DOM을 업데이트한다.
+    const oldNode = nodeToContainerMap.get(container);
+    updateElement(container, vNode, oldNode);
+  }
```

Comment:
`nodeToContainerMap.has`, `nodeToContainerMap.get`에 대해 아래와 같이 사용해 중복 코드를 줄일 수도 있을 것 같아요!

```suggestion
  const oldNode = nodeToContainerMap.get(container);

  // 최초 렌더링시에는 createElement로 DOM을 생성하고
  if (!oldNode) {
    container.append(createElement(vNode));
  } else {
    // 이후에는 updateElement로 기존 DOM을 업데이트한다.
    updateElement(container, vNode, oldNode);
  }
```

Reply:
오!! 미처 생각하지 못 했습니다! 감사합니다! 그렇게 하니까 훨씬 깔끔해지네요!👍